control_unix.go

```diff
@@ -8,13 +8,20 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// This value has been taken from https://reviews.freebsd.org/D11003 since this is not yet provided in golang.
+const FREEBSD_SO_REUSEPORT_LB = 0x00010000
+
 func Control(network, address string, c syscall.RawConn) (err error) {
 	controlErr := c.Control(func(fd uintptr) {
 		err = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEADDR, 1)
 		if err != nil {
 			return
 		}
 		err = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEPORT, 1)
+		if err != nil {
+			return
+		}
+		err = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, FREEBSD_SO_REUSEPORT_LB, 1)
```

Comment:
This should only be executed on FreeBSD. We can't just set a random sock opt on other platforms.

Reply:
Got it, I was thinking of making it future proof by having it for all unix systems. 
But if these macros are defined per OS, then it makes sense to protect it as per runtime.
Will change this.

Reply:
Done, tested on freeBSD and macOS after these changes.